lib/report.js

```diff
@@ -11,55 +11,152 @@ const line = (title, s) => {
   return `\n${lpad(title, LINE_LENGTH)}: ${s}`;
 };
 
-const list = (lines, s = '') => {
-  let key;
-  for (key in lines) s += line(capitalize(key), lines[key]);
+const list = (lines, s = '', exclude = []) => {
+  for (const key in lines) {
+    if (!exclude.includes(key)) {
+      s += line(capitalize(key), lines[key]);
+    }
+  }
   return s;
 };
 
-module.exports = (metatests) => {
-
-  metatests.results = {
-    modules: 0,
-    targets: 0,
-    functions: 0,
-    values: 0,
-    classes: 0,
-    methods: 0,
-    tests: 0,
-    errors: 0,
-  };
-
-  const renderStack = stack => {
-    const path = process.cwd();
-    const lines = stack.split('\n');
-    lines.splice(1, 1);
-    const result = lines.map((line, i) => (
-      ' '.repeat(i ? 12 : 0) +
-      line.trim().replace(/at /g, '').replace(path, '')
-    )).filter(line => !line.includes('(internal/'));
-    return result.join('\n');
-  };
-
-  metatests.report = () => {
-    const show = () => {
-      const errCount = metatests.results.errors;
-      const successed = errCount === 0;
-      metatests.results['result'] = successed ? 'PASSED' : 'FAILED';
-      console.log(
-        'Test finished with following results' +
-        list(metatests.results)
-      );
-      if (!successed) process.exit(1);
+const renderStack = stack => {
+  const path = process.cwd();
+  const lines = stack.split('\n');
+  lines.splice(1, 1);
+  const result = lines.map((line, i) => (
+    ' '.repeat(i ? 12 : 0) +
+    line.trim().replace(/at /g, '').replace(path, '')
+  )).filter(line => !line.includes('(internal/'));
+  return result.join('\n');
+};
+
+class Reporter {
+
+  constructor(test = null) {
+    this.testsCount = 0;
+    this.test = test;
+    this._testDoneCallback = t => this.record(t);
+    if (test) this.addTest(test);
+  }
+
+  addTest(test) {
+    this.testsCount++;
+    test.on('done', this._testDoneCallback);
+  }
+
+  record(/* test */) {
```

Comment:
no arguments? but in line 39 called with argument

Reply:
It has an argument, but this is a default implementation that only takes care of the machinery needed to run/report data (hence base class). Therefore this one doesn't use the argument even though it's present. So to avoid eslint error and still indicate that the argument is there I used a comment.